src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,110 @@
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
+type Props = {
+  selectedUserId: number,
+  setSelectedUserId: (userId: number) => void;
+};
+
+// eslint-disable-next-line max-len
+export const TodoList: React.FC<Props> = ({ selectedUserId, setSelectedUserId }) => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [selectQuery, setSelectQuery] = useState('');
+  const filteredTodos = todos.filter(todo => todo.title.includes(query)).filter(todo => {
+    switch (selectQuery) {
+      case 'active':
+        return todo.completed === false;
+
+      case 'completed':
+        return todo.completed === true;
+
+      default:
+        return todo;
+    }
+  });
+
+  useEffect(() => {
+    requestTodos().then(response => setTodos(response));
+  }, []);
+
+  // switch (selectQuery) {
+  //   case 'active':
+  //     setTodos(todos.filter(todo => todo.completed === false));
+  //     break;
+  //
+  //   case completed:
+  //     setTodos(todos.filter(todo => todo.completed === true))
+  //
+  //   default:
+  //
+  // }
+
+  return (
+    <div className="TodoList">
+      <h2>Todos:</h2>
+
+      <input
+        data-cy="filterByTitle"
+        type="text"
+        value={query}
+        onChange={
+          (event) => {
+            setQuery(event.target.value);
+          }
+        }
+        placeholder="Search by title"
+      />
+
+      <select value={selectQuery} onChange={(event) => setSelectQuery(event.target.value)}>
+        <option value="all">Show all</option>
+        <option value="active">Show all uncompleted</option>
+        <option value="completed">Show all completed</option>
+      </select>
+
+      <div className="TodoList__list-container">
+
+        <ul data-cy="listOfTodos" className="TodoList__list">
+          {filteredTodos?.map((todo: Todo) => (
+            <li
+              key={todo.id}
+              className={
+                todo.completed
+                  ? 'TodoList__item TodoList__item--checked'
+                  : 'TodoList__item TodoList__item--unchecked'
+              }
+            >
+              <label>
+                <input checked={todo.completed} type="checkbox" readOnly />
+                <p>{todo.title}</p>
+              </label>
+
+              <button
+                data-cy="userButton"
+                className={
+                  selectedUserId === todo.userId
+                    // eslint-disable-next-line max-len
+                    ? 'TodoList__user-button TodoList__user-button--selected button'
+                    : 'TodoList__user-button button'
+                }
+                type="button"
+                onClick={() => {
+                  if (todo.userId !== selectedUserId) {
+                    return (setSelectedUserId(todo.userId));
+                  }
+
+                  return null;
+                }}
```

Comment:
You shouldn't return anything from this function
```suggestion
                  if (todo.userId !== selectedUserId) {
                    setSelectedUserId(todo.userId);
                  }
                }}
```